shim/src/repair/functionConstructors.js

```diff
@@ -0,0 +1,68 @@
+// Adapted from SES/Caja - Copyright (C) 2011 Google Inc.
+// https://github.com/google/caja/blob/master/src/com/google/caja/ses/startSES.js
+// https://github.com/google/caja/blob/master/src/com/google/caja/ses/repairES5.js
+
+import verifyStrictFunctionBody from '../evaluate/verifyStrictFunctionBody';
+import { getPrototypeOf, setPrototypeOf, defineProperty } from '../commons';
+
+// This block replaces the original Function, %GeneratorFunction% and
+// %AsycnGeneratorFunction%, with safe implementations that preserve SES
+// confinement. After this block is done, the originals should no longer be
+// reachable.
+
+export default function repairFunctionConstructors(confinedGlobal) {
```

Comment:
I'm toying with "confinedGlobal" instead of "confinedWindow". If you agree, I will make replacements everywhere.